aviatrix/resource_aviatrix_gateway.go

```diff
@@ -1704,56 +1734,44 @@ func resourceAviatrixGatewayUpdate(d *schema.ResourceData, meta interface{}) err
 			return fmt.Errorf("failed to update Aviatrix VPN Gateway Authentication: %s", err)
 		}
 	}
-	if d.HasChange("tag_list") && (gateway.CloudType == goaviatrix.AWS || gateway.CloudType == goaviatrix.AWSGOV || gateway.CloudType == goaviatrix.AZURE) {
+
+	if d.HasChange("tag_list") || d.HasChange("tags") {
+		if !intInSlice(gateway.CloudType, []int{goaviatrix.AWS, goaviatrix.AWSGOV, goaviatrix.AZURE}) {
+			return fmt.Errorf("failed to update gateway: adding tags is only supported for AWS, AWSGOV and AZURE, cloud_type must be set to 1, 256 or 8")
+		}
+
+		_, tagListOk := d.GetOk("tag_list")
+		_, tagsOk := d.GetOk("tags")
+		if tagsOk && tagListOk {
+			return fmt.Errorf("failed to update gateway: only one of tag_list and tags is allowed at the same time")
+		}
```

Comment:
Is this check needed in the Create function as well? Or does `ConflictsWith` cover it?

Reply:
ConflictsWith covers it. I forgot to remove it after I added ConflictsWith.